charactersheet/charactersheet/services/persistence_service.js

```diff
@@ -131,6 +131,42 @@ PersistenceService.findBy = function(model, property, value) {
 };
 
 /**
+ * Given a model class, return all of the stored instances that match a given
+ * array of filter criteria.
+ *
+ * Parameters
+ * ----------
+ *
+ * model: The prototype for the type of model that is being searched for.
+ * properties: A static array of properties that the data object will be 
+ * compared against.
+ *
+ * Returns
+ * -------
+ *
+ * A list of objects of the desired type, which match the list of criteria.
+ *
+ * Usage
+ * -----
+ * ```javascript
+ *         var status = PersistenceService.findByListOfProperties(Status,
+ *           [{'name': 'characterId', 'value': key},
+ *           {'name': 'identifier', 'value': self.statusIdentifier}]);
+ * ```
+ */
+PersistenceService.findByListOfProperties = function(model, properties) {
+    return PersistenceService._findFiltered(model, function(element, idx) {
+        var count = 0;
+        properties.forEach(function(property, idx, _) {
```

Comment:
I think a better way to implement this filter is as follows: 

```js
// Ensure that every comparison returns true.
return properties.every(function(element, idx, _) {
  return element[property.name] === property.value;
});
```
I think this is much more readable and less imperative. Thoughts?

[Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)